cmd/microcloud/ask.go

```diff
@@ -1096,6 +1096,71 @@ func (c *initConfig) askOVNNetwork(sh *service.Handler) error {
 		}
 	}
 
+	canOVNUnderlay := true
+	for peer, state := range c.state {
+		if len(state.AvailableOVNInterfaces) == 0 {
+			fmt.Printf("Not enough interfaces available on %s to create an underlay network, skipping\n", peer)
+			canOVNUnderlay = false
+			break
+		}
+	}
+
+	var ovnUnderlaySelectedIPs map[string]string
+	if canOVNUnderlay {
+		// TODO: Call `sh.Services[types.MicroOVN].(*service.OVNService).SupportsFeature(context.Background(), "custom_encapsulation_ip")`
+		// when MicroCloud will be updated with microcluster/v2
+
+		wantsDedicatedUnderlay, err := c.asker.AskBool("Configure dedicated underlay networking? (yes/no) [default=no]: ", "no")
+		if err != nil {
+			return err
+		}
+
+		if wantsDedicatedUnderlay {
+			header = []string{"LOCATION", "IFACE", "TYPE", "IP ADDRESS (CIDR)"}
+			fmt.Println("Select exactly one network interface from each cluster member:")
+			data = [][]string{}
+			for peer, state := range c.state {
+				for _, net := range state.AvailableOVNInterfaces {
+					for _, addr := range net.Addresses {
+						data = append(data, []string{peer, net.Network.Name, net.Network.Type, addr})
+					}
+				}
+			}
```

Comment:
`c.state` is a list of all existing and prospective cluster members, which means the table will show entries for nodes that may have been already configured during `microcloud init`. Since the config is only applied on bootstrap/join, the new configuration will be ignored.

If you switched this to `c.systems`, you would instead iterate over only the new systems. But that would introduce the opposite problem when running `microcloud add` of what to do if the user initially selected "no" for underlay network setup when they ran `microcloud init`. 

Ideally, we should be iterating over the list of all systems (existing or new) which have not set a geneve address. Problem is I don't see a way to fetch the geneve address from MicroOVN over the API. 

Could you speak to the MicroOVN team about adding a way to query the nb/sb databases over the API? Or at least add the geneve address to `microovn status`, since it's included as a config in `microovn init`?

Reply:
Is there an option for microcloud to query OVN directly? MicroOVN provides [content interfaces](https://github.com/canonical/microovn/blob/0dc300879ab2063d0531811c15325217e8f3bcc6/snap/snapcraft.yaml#L15-L30) that should enable you to talk directly to the OVN/OVS. If you need overall cluster information, you'd need to query equivalent of
```
microovn.ovn-sbctl list encap
```
Example output:
```
_uuid               : a3096d41-e53e-422a-ba50-2fec3542d9ad
chassis_name        : movn1
ip                  : "10.75.224.231"
options             : {csum="true"}
type                : geneve
```

If you are interested only about the local chassis, you could run
```
microovn.ovs-vsctl get open_vswitch . external_ids:ovn-encap-ip
```
Example output:
```
"10.75.224.231"
```

Reply:
@gabrielmougard so in the loop above, you should iterate on `c.systems` instead of `c.state`. 

Also, we need to skip underlay configuration during `microcloud service add` because in that case `c.state == c.systems`. So you'll have to check if they are already clustered:
```go
for peer, systems := range c.systems {
   // skip any systems that have already been clustered, but are available for other configuration.
   if c.state[c.name].ExistingClusters[types.MicroOVN][peer] == "" {
        continue
    }
}
```

Reply:
> @gabrielmougard so in the loop above, you should iterate on `c.systems` instead of `c.state`.
> 
> Also, we need to skip underlay configuration during `microcloud service add` because in that case `c.state == c.systems`. So you'll have to check if they are already clustered:
> 
> ```go
> for peer, systems := range c.systems {
>    // skip any systems that have already been clustered, but are available for other configuration.
>    state, ok := c.state[c.name] 
>    if ok {
>       if state.ExistingClusters[types.MicroOVN][peer] != "" {
>          continue
>       }
>    }
> }
> ```

should be fixed now @masnax